shared/common-adapters/tab-bar.js.flow

```diff
@@ -4,41 +4,40 @@ import React, {Component} from 'react'
 import Avatar from './avatar'
 import type {Props as IconProps} from './icon'
 
-export type ItemProps = {
-  tabBarButton?: React$Element,
-  label?: string,
+export type Props = $Shape<{
   style?: Object,
-  selected?: boolean,
-  selectedColor?: string,
-  onClick?: () => void,
-  styleContainer?: Object,
-  styleLabel?: Object,
-  children?: React$Element
-}
-
-export class TabBarItem extends Component<void, ItemProps, void> {
-}
-
-export type Props = {
-  style?: Object,
-  tabBarStyle?: Object,
+  styleTabBar?: Object,
   children?: Array<React$Element<ItemProps>>,
   tabBarOnBottom?: boolean,
-  tabWidth?: number,
   underlined?: boolean
-}
+}>
+
+export default class TabBar extends Component<void, Props, void> { }
+
+export type ItemProps = $Shape<{
+  tabBarButton: React$Element,
+  label: string,
+  selected: boolean,
+  selectedColor: string,
+  onClick: () => void,
+  style: Object,
+  styleContainer: Object,
+  children: React$Element
+}>
 
-export type TabBarButtonProps = {
+export class TabBarItem extends Component<void, ItemProps, void> { }
+
+export type TabBarButtonProps = $Shape<{
   selected?: boolean,
-  source: {type: 'icon', icon: IconProps.type, label?: string} | {type: 'avatar', avatar: Avatar},
+  source: {type: 'icon', icon: IconProps.type} | {type: 'avatar', avatar: Avatar},
```

Comment:
I know this wasn't you, but could you use `IconType` here (imported from the same file as `IconProps`)